include/range/v3/range/access.hpp

```diff
@@ -57,11 +66,10 @@ namespace ranges
         // clang-format off
         template<typename T>
         CPP_concept_bool has_member_begin =
-            CPP_requires ((T &) t) //
+            CPP_requires ((T &&) t) //
             (
-                _begin_::is_iterator(t.begin())
-            ) &&
-            std::is_lvalue_reference<T>::value;
+                _begin_::is_iterator(CPP_fwd(t).begin())
+            );
```

Comment:
Shouldn't we be simplifying `has_non_member_begin` to work with lvalues rather than adding perfect-forwarding to `has_member_begin`?

Reply:
I'm not sure I understand the question. The intent is to forward the range expression through both cases.

Reply:
I think @CaseyCarter is expecting us to promote rvalues to lvalues. I don't think that's necessary, and I like @BRevzin's formulation. Casey, is there a reason why we want to be promoting rvalues to lvalues?

Reply:
I'm envisioning a design more like the pre-`forwarding-range` design in which the CPOs always dispatch lvalue arguments to the "hook" functions so implementors of range types - even `safe_range` types - need not worry about accepting rvalues in `begin` and `end`. I'm not sure *why* I'm envisioning that, given that every model of `safe_range` I've ever seen has member `begin`/`end` that are callable on rvalues. 

I, uh, want to eliminate the runtime cost of all of the xvalue casts. Yeah, that's it - those crazy expensive casts.